### ISSUE TITLE
cumsum() on 1D tensor requires the dimension argument

### ISSUE THREAD

**Issue by jucor** [torch/DEPRECEATED-torch7-distro#81]
As mentioned to Koray, when using cumsum() on a 1D tensor, torch generates an error if the dimension argument is not given -- whereas doc states it should operate on its first dimension (which is, here, the only dimension anyways). See below. 

```
Torch 7.0  Copyright (C) 2001-2011 Idiap, NEC Labs, NYU
xLua 5.1  Copyright (C) 1994-2008 Lua.org, PUC-Rio
t7> x=torch.Tensor({1,2,3})
t7> =x:cumsum()
[string "return x:cumsum()"]:1: expected arguments: [*DoubleTensor*] DoubleTensor index
stack traceback:
    [C]: in function 'cumsum'
    [string "return x:cumsum()"]:1: in main chunk
    [C]: ?
t7> return x:cumsum(1)
 1
 3
 6
[torch.DoubleTensor of dimension 3]
```

Thanks for the great work!

**Comment by andresy** [comment]
Ach, I think we (I?) changed the internal of this some time ago already.
It is actually for any kind of tensor dimension, not only for 1D.
If it had to choose one dimension, I supposed it should choose the last one anyways -- the one which is contiguous.

I thought it was more clear to specify it -- to be compatible with sum() for e.g., which makes a clear difference between having an extra index or not.

I do know what is best -- should we only change the doc?

**Comment by jucor** [comment]
Thanks for the fast answer! Mmmm, I see the catch with sum(). Tricky semantic choice. Would it make sense to have cumsum() require the dimension for 2-D vectors and above, but not for 1-D vectors where it would accept to default to 1 ? Or would that be counter-intuitive ?

**Comment by andresy** [comment]
It is a tough decision. If you consider select(), it will return you a number if the dimension is 1 and [a tensor with 1 dimension removed, if the dimension is > 1].

We had in the past a discussion about what we should do for this particular sum(), cumsum()... etc... functions. Right now they do not remove any dimension -- in that respect, they behave a bit like narrow(). (So if you sum over dimension "i", the resulting tensor size(i) will be 1). Can't remember right now why we decided to keep it like that, have to discuss it again for the next torch version.

Note: we added :squeeze() and squeeze1d() to remove dimensions = 1 some time ago.
